usaspending_api/search/filters/time_period/query_types.py

```diff
@@ -58,3 +58,52 @@ def _return_date_type(self) -> str:
         if ret_date_type in self._date_type_transaction_search_map:
             ret_date_type = self._date_type_transaction_search_map[ret_date_type]
         return ret_date_type
+
+
+class AwardSearchTimePeriod(AbstractTimePeriod):
+    """A time period implementation that's designed to suit the time period filters
+    on award search.
+
+    This concrete implementation supports the filter [AwardSearchTimePeriodObject](#usaspending_api/api_contracts/search_filters.md#award-search-time-period-object).
+    """
+
+    def __init__(self, default_start_date: str, default_end_date: str):
+        """Constructor to create AwardSearchTimePeriod object.
+
+        Args:
+            default_start_date: This is the start date value to use when start date is not present in the filter value.
+                See start date in [AwardSearchTimePeriodObject](#usaspending_api/api_contracts/search_filters.md#award-search-time-period-object)
+                for definition.
+            default_end_date: This is the end date value to use when end date is not present in the filter value.
+                See end date in [AwardSearchTimePeriodObject](#usaspending_api/api_contracts/search_filters.md#award-search-time-period-object)
+                for definition.
+        """
+        self._default_start_date = default_start_date
+        self._default_end_date = default_end_date
+        self._filter_value = {}
+
+    @property
+    def filter_value(self):
+        return self._filter_value
+
+    @filter_value.setter
+    def filter_value(self, filter_value):
+        self._filter_value = filter_value
+
+    def start_date(self):
+        return self._filter_value.get("start_date") or self._default_start_date
+
+    def end_date(self):
+        return self._filter_value.get("end_date") or self._default_end_date
+
+    def gte_date_type(self):
+        return self._filter_value.get("date_type", "action_date")
```

Comment:
Why use `action_date` here? For `new_awards_only`, wouldn't we want to use `date_signed` for both the upper and lower comparisons?

Reply:
What do you mean. We are not using `action_date` for `new_awards_only`. Nothing in this class has knowledge of new awards only. You need to go look at the new awards only decorator. Everything "new award only" related is encapsulated by that class.

Reply:
Ah right, I must have been thinking about that decorator when I wrote this comment. Good to go here.